File: app/src/main/java/com/github/alexeygorovoy/moxytemplate/dagger/demo/details/HeroDetailsComponent.java
package com.github.alexeygorovoy.moxytemplate.dagger.demo.details;

import com.github.alexeygorovoy.moxytemplate.ui.demo.details.view.HeroDetailsFragment;

import dagger.Subcomponent;


@Subcomponent(modules = {HeroDetailsModule.class})
public interface HeroDetailsComponent {
    void inject(HeroDetailsFragment fragment);
}

File: app/src/main/java/com/github/alexeygorovoy/moxytemplate/dagger/demo/details/HeroDetailsComponent.kt
package com.github.alexeygorovoy.moxytemplate.dagger.demo.details

import com.github.alexeygorovoy.moxytemplate.ui.demo.details.view.HeroDetailsFragment

import dagger.Subcomponent

@Subcomponent(modules = [HeroDetailsModule::class])
interface HeroDetailsComponent {
    fun inject(fragment: HeroDetailsFragment)
}
